roadmap/implementers-guide/src/node/dispute/participation.md

```diff
@@ -69,3 +81,14 @@ Nothing to store
   1. includes `CandidateReceipts`
   1. includes `Vote`s
 1. Delete all relevant data associated with this dispute
+
+### Blacklisting bad block and invalid decendants
+
+In case a block was disputed successfully, and is now deemed invalid.
+
+1. Query `ChainApiMessage::Descendants` for all descendants of the disputed `Hash`.
+1. Blacklist all descendants of the disputed block `ChainApiMessage::Blacklist`.
+1. Check if the dispute block was finalized
+1. iff:
+    1. put the chain into governance mode
+1. Craft and enqueue an unsigned transaction to reward/slash the validator of the proof, that is part of the incoming message, directly.
```

Comment:
Is this not something the runtime should handle automatically?

Reply:
I am not quite sure on where that logic should live. Technically it's possible from both sides. But do we need that to be updatable? I had the impression that was rather static and essentially a candidate hash and a list of candidate receipts to be included in the inherent of that transaction.

Reply:
This is referring to slashing validators, though. The dispute logic on-chain should have an idea of which disputes have concluded and how all the validators involved voted. Therefore it should be able to perform slashing automatically.